tools/swift-stdlib-tool/swift-stdlib-tool.cpp

```diff
@@ -670,583 +643,576 @@ -(NSString *)sst_command {
 
 
 // This executable's own path.
-NSString *self_executable = []() -> NSString * {
+std::filesystem::path self_executable = []() -> std::filesystem::path {
     uint32_t len = 0;
-    _NSGetExecutablePath(nil, &len);
+    _NSGetExecutablePath(NULL, &len);
     std::vector<char> buffer;
     buffer.reserve(len);
     _NSGetExecutablePath(buffer.data(), &len);
```

Comment:
Will `_NSGetExecutablePath` fill in the nul byte at the end? Even if so, I would feel more comfortable if we constructed a std::span with the `len` here and converted that to a `std::filesystem::path`.

Reply:
It seems like `std::span` was introduced in C++20. `_NSGetExecutablePath` returns a string with a size of `MAXPATHLEN` at most so I was able to remove the first call. It does fill the nul byte at the end but I'm zeroing anyways.